src/main/k/in-progress/tests/list-unify-3/list-unify-3.k

```diff
@@ -0,0 +1,19 @@
+module LIST-UNIFY-3
+  
+  imports DOMAINS
+  
+  syntax Val ::= "a" | "b" | "c"
+
+  configuration
+    <T>
+      <k> $PGM:Val </k>
+      <l> ListItem(a) ListItem(b) ListItem(c) </l>
+      <v> .K </v>
+    </T>
+
+  rule 
+    <k> V:Val </k>
+    <l> ... ListItem(V) ... </l>
```

Comment:
Actually I don't think we're aiming on handling this for now.  Maybe you could make two tests, one with matching at one end, one with matching at the other.

Reply:
I added two more tests very similar to this one, to try matching on the left and right.